black.py

```diff
@@ -698,6 +715,13 @@ def maybe_decrement_after_lambda_arguments(self, leaf: Leaf) -> bool:
 
         return False
 
+    def get_open_lsqb(self, leaf: Leaf) -> Optional[Leaf]:
+        """Returns the most recent opening square bracket at `leaf` (if any)."""
```

Comment:
1. Adding a `leaf` argument to `get_open_lsqb()` creates a rather bizarre signature. It suggests that `leaf`'s position on the line has something to do with the result but it doesn't. It's simply a special case if `leaf` itself is a LSQB and is not otherwise used.  I think that path should be handled directly in `is_complex_subscript()`.

2. nit: docstrings should be in imperative: "Return" rather than "Returns".